src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_map.c

```diff
@@ -1,108 +1,918 @@
+
+            pauseCtx->mapPageVtx[122].v.ob[1] = pauseCtx->mapPageVtx[123].v.ob[1] =
+                pauseCtx->mapPageVtx[120].v.ob[1] - 8;
+        }
+
+        // Loop over RegionId (i), unused vtxIndex (j), unused (k)
+        for (i = 0, j = 0; i < REGION_MAX; i++, k++, j += 4) {
+            if (pauseCtx->worldMapPoints[i]) {
+                gSPVertex(POLY_OPA_DISP++, &pauseCtx->mapPageVtx[120 + i * 4], 4, 0);
+                gSP1Quadrangle(POLY_OPA_DISP++, 0, 2, 3, 1, 0);
+            }
+        }
+
+    } else {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetCombineMode(POLY_OPA_DISP++, G_CC_MODULATEIA_PRIM, G_CC_MODULATEIA_PRIM);
+        gDPLoadTextureBlock(POLY_OPA_DISP++, gWorldMapOwlFaceTex, G_IM_FMT_RGBA, G_IM_SIZ_32b, 24, 12, 0,
+                            G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMIRROR | G_TX_WRAP, G_TX_NOMASK, G_TX_NOMASK, G_TX_NOLOD,
+                            G_TX_NOLOD);
+
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, pauseCtx->alpha);
+
+        if (R_PAUSE_DBG_MAP_CLOUD_ON) {
+            gSaveContext.save.worldMapCloudVisibility |= (u16)~0x8000;
+
+            pauseCtx->mapPageVtx[164].v.ob[0] = pauseCtx->mapPageVtx[166].v.ob[0] = R_PAUSE_DBG_MAP_CLOUD_X;
+
+            pauseCtx->mapPageVtx[165].v.ob[0] = pauseCtx->mapPageVtx[167].v.ob[0] =
+                pauseCtx->mapPageVtx[164].v.ob[0] + 24;
+
+            pauseCtx->mapPageVtx[164].v.ob[1] = pauseCtx->mapPageVtx[165].v.ob[1] = R_PAUSE_DBG_MAP_CLOUD_Y;
+
+            pauseCtx->mapPageVtx[166].v.ob[1] = pauseCtx->mapPageVtx[167].v.ob[1] =
+                pauseCtx->mapPageVtx[164].v.ob[1] - 12;
+        }
+
+        // Loop over OwlWarpId (i), unused vtxIndex (j), unused (k)
+        for (i = 0, j = 0; i < OWL_WARP_ENTRANCE; i++, k++, j += 4) {
+            if (pauseCtx->worldMapPoints[i]) {
+                gSPVertex(POLY_OPA_DISP++, &pauseCtx->mapPageVtx[164 + i * 4], 4, 0);
+                gSP1Quadrangle(POLY_OPA_DISP++, 0, 2, 3, 1, 0);
+            }
+        }
+    }
+
+    if ((pauseCtx->pageIndex == PAUSE_MAP) && (pauseCtx->unk_200 == 0) && (pauseCtx->state == 6) &&
+        (pauseCtx->state != 7) && !((pauseCtx->state >= 8) && (pauseCtx->state <= 0x12))) {
+        j = 0;
+        n = 0;
+        sceneId = play->sceneId;
+        if (sceneId == SCENE_KAKUSIANA) { // Lone Peak Shrine & Grottos
+            if (play->roomCtx.curRoom.num == 5) {
+                sceneId = SCENE_11GORONNOSATO; // Goron Village (winter)
+            } else if ((play->roomCtx.curRoom.num == 6) || (play->roomCtx.curRoom.num == 8) ||
+                       (play->roomCtx.curRoom.num == 12)) {
+                sceneId = SCENE_22DEKUCITY; // Deku Palace
+            } else {
+                sceneId = Entrance_GetSceneIdAbsolute(((void)0, gSaveContext.respawn[RESPAWN_MODE_UNK_3].entrance));
+            }
+        }
+
+        // Loop over regionId (n) and regionIdIndex (j)
```

Comment:
Is the purpose of this to highlight the player's current region or something? I think it needs a comment.

Reply:
Yeah, I double checked and this loop is searching for the regionId that Link is currently in, so that after this loop, it can draw Link's Head at the right region

![image](https://user-images.githubusercontent.com/47598039/193350292-b05694ac-b724-4236-b4fb-69970ddb4d85.png)